Marlin/src/pins/pinsDebug_list.h

```diff
@@ -2300,21 +2300,36 @@
 //
 #if PIN_EXISTS(SDIO_D0)
   REPORT_NAME_DIGITAL(__LINE__, SDIO_D0_PIN)
+#elif defined(BOARD_SDIO_D0)
+  REPORT_NAME_DIGITAL(__LINE__, BOARD_SDIO_D0)
```

Comment:
Only pins that are actually assigned to a Marlin function need to be reported here. We don't need to report pins defined with a `BOARD_` prefix, as they are merely intended as defaults if not otherwise defined.

Reply:
Currently the SDIO pins are defined by the HAL and then in some instances pins files define some `BOARD_SDIO_*` pins (without a `_PIN` suffix). These break the pattern that we had previously established, so it would be good to fix these up. The way it should work is:
- Define `BOARD_SDIO_*_PIN` pins in any pins files that want to override the HAL.
- Define `SDIO_*_PIN` as `BOARD_SDIO_*_PIN` when such a define exists,
- Otherwise, define `SDIO_*_PIN` based on the MCU.
I see that some of these are defined by `variants/*/board.h` while only `pins_AQUILA_101.h` is the only pins file that does so. It's not entirely clear where in the build process `variants/*/board.h` is included, but I presume that it's ahead of all Marlin sources since it's part of the platform support. So, the common approach should work as described above for all the `SDIO_*_PIN` defines.

Reply:
That said, caveats like this should not hold up this PR. Any items in this sweeping PR that are still under discussion can be moved to a separate PR as necessary.

Reply:
I thought I had since reverted this. so I'll go through this once over to see if there is anything left to do. otherwise if you can point anything out, or list it what needs to be changed lmk